discord.go

```diff
@@ -0,0 +1,272 @@
+	message, err := Discord.session.ChannelMessageSend(channelID, content)
+	if err != nil {
+		return nil, fmt.Errorf("failed to send message to primary channel")
+	}
+	Discord.logger.WithFields(log.Fields{
+		"author":    message.Author,
+		"channelID": message.ChannelID,
+		"guildID":   message.GuildID,
+		"message":   message.Content,
+		"dateTime":  time.Now().String(),
+	})
+	return message, nil
+}
+func (Discord *DiscordImpl) SendDM(userID string, message string) (*discordgo.Message, error) {
+	channel, err := Discord.session.UserChannelCreate(userID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create private message channel with userID: %s", userID)
+	}
+	return Discord.SendChannelMessage(channel.ID, message)
+}
+
+func PanicAlertCallback(s *discordgo.Session, i *discordgo.InteractionCreate) {
+	// TODO write logic for starting a panicalert vote
+	// TODO if enough votes then call SendDM method passing the information from the config.ContactOnVote {Discord {}} struct
+	// TODO if enough votes then call Twilio API to text/call the number from the config.ContactOnVote {Twilio {}} struct
+	// TODO if enough votes then call Email handler to email the addresses from the config.ContactOnVote {Email {}} struct
+	// TODO write logic for if vote fails. No one is contacted but perhaps a message is sent to the PrimaryChannel. Use SendChannelMessage
+}
+
+func PanicBanCallback(s *discordgo.Session, i *discordgo.InteractionCreate) {
+	// TODO write logic for starting a panicban vote
+	// TODO if enough votes then call SendDM method passing the information from the config.ContactOnVote {Discord {}} struct
+	// TODO if enough votes then call Twilio API to text/call the number from the config.ContactOnVote {Twilio {}} struct
+	// TODO if enough votes then call Email handler to email the addresses from the config.ContactOnVote {Email {}} struct
+	// TODO if enough votes then call BanUser method
+	// TODO write logic for if vote fails. No one is contacted but perhaps a message is sent to the PrimaryChannel. Use SendChannelMessage
+}
+
+func EmbedReactionCallback(s *discordgo.Session, i *discordgo.InteractionCreate) {
+	// TODO use this for whenever we recieve a reaction to a panicalert / panicban
+	// This function will be used to tally up the votes and then take action.
+}
+
+type DiscordImpl struct {
+	botToken              string
+	guildID               string
+	primaryChannelID      string
+	logger                *log.Logger
+	session               *discordgo.Session
+	embedReactionCallback func(s *discordgo.Session, i *discordgo.InteractionCreate)
+	panicAlertCallback    func(s *discordgo.Session, i *discordgo.InteractionCreate)
+	panicBanCallback      func(s *discordgo.Session, i *discordgo.InteractionCreate)
+}
+
+type DiscordImplArgs struct {
+	BotToken              string
+	GuildID               string
+	PrimaryChannelID      string
+	Logger                *log.Logger
+	Session               *discordgo.Session
```

Comment:
This is not needed as we build the session in new

Reply:
Do we not want to store the session we create inside of this struct so we can use it elsewhere in the package?

See inside the method registerSlashCommands on line 229. Where I need access to the session to add the event listeners using the AddHandler method available only on *discordgo.Session

Reply:
Poor explanation on my part. I'm having you move all setup of the bot into the new Discord call,  which will return the interface that only has those 3 methods,  and only calls the mentioned callbacks. This means that session will not be needed outside.